### PR TITLE
completed

### PR DESCRIPTION
**Pull request by 1102-Briski-Sean**
# Submitting Week 5 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [X] Career Readiness
- [X] Coding with AI
- [X] Interview Questions
- [X] JavaScript Challenges
- [X] Ruby Challenges

### ✋ Reminder

Do NOT merge your assessment!

### 📝 Feedback

If you would like specific feedback please leave details here:

### CODE REVIEW

File: coding-with-ai.rb [hunk 1]
```diff
@@ -6,11 +6,55 @@
...
+
+# My Final Thoughts on Chat GPT's Contribution:
+# It was nice to see that my initial interpretation of the code was very similar to Chat GPT's interpretation of the code.
+# It is interesting to see the AI emphasize what it thinks is vital for the code to work as intended.
+# Chat GPT's prompt made sure to mention that the votes need to be case-insensitive. This was a detail I failed to include in either of the two prompts I created on my own. In my prompts, I specifically requested the output to be in the form of a Hash. Chat GPT only required that the total votes for each type of taco should be returned. The AI did not make any distinction as to how this information should be presented.
+
+#_____________________Final Prompt________________________
+  # Create a function that takes in an Array of taco votes and returns the total number of votes received for each type of taco. The vote counts need to be case-insensitave as votes can be received in lowercase, uppercase, or mixed case.
```
Comment (elysemontano) [inline coding-with-ai.rb:60]: Excellent work in this section!!!  👏

File: interview-questions.md [hunk 2]
```diff
@@ -8,32 +8,47 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+    ]
+
+Arrays are collections of values. Hashes are collections of key: value pairs. Values in an array can be accessed by using their index. Values in a hash can be accessed by calling on their key. Arrays can be flattened. Arrays are ideal for isolating data of a similar type. Hashes are ideal for collecting and analyzing more complex combinations of data.
+
+Researched answer: The output of a hash in the terminal uses an older syntax for writing hashes. For this older syntax, the colon is on the left side of the key and a hash rocket separates the key from its value. Arrays are bounded by square brackets. For hashes, key: value pairs for an item are bounded by curly brackets. If there are multiple items in a hash, this collection of items is bounded by square brackets similar to an array.
 
 2. What does it mean to implement a TTD workflow?
 
-Your answer:
+Your answer: I know TDD is Test-Driven Development. I do not know what TTD is.
```
Comment (elysemontano) [inline interview-questions.md:26]: 🤣 ...Whoops!  I do appreciate your humor, I would keep in mind though that if you were in an interview and they use an inaccurate term, be appropriate and ask if they mean something different than what they initially said to clarify.  This situation can actually happen especially if you are talking to an HR representative who is doing an initial screening and knows nothing about code, instead are just reading off of a prompt.

File: javascript-challenges.test.js [hunk 3]
```diff
@@ -38,6 +95,51 @@ const hand3 = [5, 5, 5, 5, 4]
 const hand4 = [7, 2, 7, 2, 7]
 // Expected output: true
 
+// Added additional tests for the cards represented by letters
+const hand5 = ['A', 'K', 'K', 'A', 'K']
+// Expected output: true
+const hand6 = [7, 'A', 'K', 7, 'K']
+// Expected output: false
+const hand7 = [6, 'Q', 'Q', 6, 6]
```
Comment (elysemontano) [inline javascript-challenges.test.js:103]: Outstanding work on this problem!  I really like that you also added your own additional test cases as well! 🎉

File: ruby-challenges.rb [hunk 4]
```diff
@@ -5,15 +5,59 @@
...
 
 # --------------------2a) Create a class called Bike that is initialized with a model, wheels, and current_speed. The default number of wheels is 2. The current_speed should start at 0. Create a bike_info method that returns a sentence with all the data from the bike object.
 
 # Expected output example: 'The Trek bike has 2 wheels and is going 0 mph.
 
 # Pseudo code:
+# arguments: model, wheels
+# default values: wheels = 2
+# instance variables: model, wheels, current_speed
+# bike_info method: String interpolation with all data from bike object
```
Comment (elysemontano) [inline ruby-challenges.rb:31]: Your pseudocode could be expanded a little more here, but otherwise excellent work in this section!

**Review by elysemontano — APPROVED** [review]
Awesome work once again Sean!  This assessment displays a competent skill acquisition level.  I left a few comments throughout to take a look at.  Your approach to your tests were fantastic in the coding challenges, keep up that process of thinking of those edge cases and how to test for them.  That will continue to grow you as a developer.

**Comment by 1102-Briski-Sean** [comment]
Thank you for your feedback.
